src/plugins/intel_cpu/src/transformations/cpu_opset/arm/pass/convert_reduce_no_keep_dims.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2020-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "convert_reduce_no_keep_dims.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/opsets/opset8.hpp"
+
+template <class T>
+ov::matcher_pass_callback ov::intel_cpu::ConvertReduceNoKeepDimsBase::convert_reduce() {
+    return [&](ov::pass::pattern::Matcher& m) {
+        auto reduce = std::dynamic_pointer_cast<T>(m.get_match_root());
+        if (!reduce || reduce->is_dynamic() || reduce->get_keep_dims()) {
```

Comment:
Show we really not allow all the dynamic reduces?
What if the reduction axis is not dynamic, the transformation should be applicable for such cases, I guess.
Can we cover this in the single layer test by the way?

Reply:
Updated transformation to accept dynamic shapes and updated unit test accordingly.